pkg/teleworker/teleworker.go

```diff
@@ -0,0 +1,89 @@
+package teleworker
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	api "github.com/spirifoxy/teleworker/internal/api/v1"
+)
+
+func (j *Job) Start() error {
+	j.mu.Lock()
+	defer j.mu.Unlock()
+
+	if j.state.Status != api.JobStatus_STARTING {
+		return fmt.Errorf("not possible to start the job: unexpected status %s on start", j.state.StatusName())
+	}
+
+	err := j.cmd.Start()
+	if err != nil {
+		return fmt.Errorf("error starting the task: %w", err)
+	}
+	j.state.Status = api.JobStatus_ALIVE
+
+	go j.wait()
+	return nil
+}
+
+func (j *Job) wait() {
+	err := j.cmd.Wait()
+	exitCode := j.cmd.ProcessState.ExitCode()
+	if err != nil && exitCode != -1 {
+		j.state.ExitErr = err
+	}
+
+	j.state.ExitCode = exitCode
+	j.state.Status = api.JobStatus_FINISHED
```

Comment:
I think you have a race here because you're accessing `j.state` here not under a lock and also accessing it in `Status`.

Reply:
Yeah, you are right, I missed locking in the `wait` call, thanks.
As loggers have their own locks, I'm now adding it only for the `j.state` update, not the whole method